src/ScheduleService.Business/Category/CreateCategoryCommand.cs

```diff
@@ -15,26 +20,50 @@ public class CreateCategoryCommand : ICreateCategoryCommand
 {
   private readonly IHttpContextAccessor _httpContextAccessor;
   private readonly ICategoryRepository _repository;
+  private readonly IDbCategoryMapper _mapper;
   private readonly ICreateCategoryRequestValidator _validator;
   private readonly IResponseCreator _responseCreator;
   private readonly IAccessValidator _accessValidator;
 
   public CreateCategoryCommand(
     ICategoryRepository repository,
+    IDbCategoryMapper mapper,
     IAccessValidator accessValidator,
     IHttpContextAccessor httpContextAccessor,
     IResponseCreator responseCreator,
     ICreateCategoryRequestValidator validator)
   {
     _httpContextAccessor = httpContextAccessor;
     _repository = repository;
+    _mapper = mapper;
     _accessValidator = accessValidator;
     _responseCreator = responseCreator;
     _validator = validator;
   }
 
-  public Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateCategoryRequest request)
+  public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateCategoryRequest request)
   {
-    throw new NotImplementedException();
+    ValidationResult validationResult = await _validator.ValidateAsync(request);
+
+    if (!validationResult.IsValid)
+    {
+      return _responseCreator.CreateFailureResponse<Guid?>(
+        HttpStatusCode.BadRequest,
+        validationResult.Errors.Select(v => v.ErrorMessage).ToList());
+    }
+
+    OperationResultResponse<Guid?> response = new();
+    Guid createdBy = _httpContextAccessor.HttpContext.GetUserId();
+
+    response.Body = await _repository.CreateAsync(_mapper.Map(request, createdBy));
```

Comment:
You may merge 55 and 58 line
{
    Body = await _repository.CreateAsync(_mapper.Map(request, createdBy))
}

Reply:
done